src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1334,6 +1334,32 @@ private void WriteResourceAssemblies()
                     {
                         WriteMetadata(MetadataKeys.AssetType, "resources");
                         string locale = asset.Properties["locale"];
+                        // Locales from packages can be free-form, so we normalize them to the standard
+                        // forms here. If the locale is mixed-case, that can cause issues on case-sensitive
+                        // file systems when the locale-specific assets are copied.
+                        try
+                        {
+                            var normalizedLocale = System.Globalization.CultureInfo.GetCultureInfo(locale).Name;
+                            _task.Log.Log(new (MessageLevel.HighImportance, 
+                                string.Format(Strings.PackageContainsIncorrectlyCasedLocale,
+                                    package.Name, package.Version.ToNormalizedString(), locale, normalizedLocale
+                                ),
+                                "NETSDK1187"));
```

Comment:
It doesn't look like this checks to see whether the locale is actually incorrectly cased before emitting the warning.

Also, we should prefer calling LogWarning, which parses out the error code for us.

```suggestion
                            _task.Log.LogWarning(Strings.PackageContainsIncorrectlyCasedLocale,
                                    package.Name, package.Version.ToNormalizedString(), locale, normalizedLocale);
```